packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -509,28 +512,51 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         // only if it is first failure, otherwise fallback to get call.
         if (usePost) {
             try {
-                return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, true);
+                const odspSnapshot = await this.fetchSnapshotWithTimeout(snapshotOptions, tokenFetchOptions, true);
+                return odspSnapshot;
             } catch (error) {
                 const errorType = error.errorType;
                 if ((errorType === DriverErrorType.authorizationError || errorType === DriverErrorType.incorrectServerResponse) && tokenFetchOptions.refresh === false) {
                     throw error;
                 }
                 this.logger.sendErrorEvent({ eventName: "TreeLatest_FallBackToGetRequest" }, error);
-                return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, false);
+                return this.fetchSnapshotWithTimeout(snapshotOptions, tokenFetchOptions, false);
             }
         } else {
-            return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, false);
+            return this.fetchSnapshotWithTimeout(snapshotOptions, tokenFetchOptions, false);
+        }
+    }
+
+    private async fetchSnapshotWithTimeout(snapshotOptions: ISnapshotOptions, tokenFetchOptions: TokenFetchOptions, usePost: boolean) {
```

Comment:
is  `usePost` what controls blob vs non-blob? seems weird.

Reply:
No, usePost is another feature where we use POST network request to fetch snapshot instead of GET.

Reply:
found it

Reply:
+1 I haven't looked at this code before but I think it's generally a good principal to reflect what you expect the caller cares about.

Ie. they probably don't care if the request uses post vs get but they might want blob or non-blob.

Reply:
Actually this is trees latest call and only the driver decides currently whether to fetch with blobs or not. So it is internal to driver. Maybe in future we add a policy to specify that also.